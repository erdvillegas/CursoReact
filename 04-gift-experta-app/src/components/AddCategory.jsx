import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    // Recibo el valor del input
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };
    
    const onSubmit = ( event ) => {
        
        event.preventDefault();
        const newInputValue = inputValue.trim();

        if (newInputValue.length <= 1) return;

        onNewCategory(newInputValue);
        setInputValue('');
    };

    return (
        <form onSubmit={ onSubmit }>
            <input
                type="text"
                title="Nueva Categoria"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
