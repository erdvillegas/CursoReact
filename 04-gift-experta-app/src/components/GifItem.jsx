import React from 'react'

const GifItem = ({ title, url, id}) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
    </div>
  )
}

export default GifItem
