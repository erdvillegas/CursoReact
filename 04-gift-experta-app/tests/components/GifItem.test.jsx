import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem>', () => {
  
    const title = "Link";
    const url = 'https://www.16personalities.com/static/images/personality-types/avatars/infj-advocate.png';

    test('debe hacer match con el snapshopt', () => {
        const { container } =  render(<GifItem title={title} url={url} />);
        expect( container ).toMatchSnapshot();
    });
    
    test('debe de mostrar la imagen con el URL y el ALT indicado ', () => {
        
        const { container } =  render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('debe de mostrar el titulo en el componente', () => {
      
        const { container } =  render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title )).toBeTruthy();
    });
    
});
