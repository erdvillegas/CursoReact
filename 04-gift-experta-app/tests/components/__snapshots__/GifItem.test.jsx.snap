// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> debe hacer match con el snapshopt 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Link"
      src="https://www.16personalities.com/static/images/personality-types/avatars/infj-advocate.png"
    />
    <p>
      Link
    </p>
  </div>
</div>
`;
